roofit/roofitcore/inc/RooAbsReal.h

```diff
@@ -463,7 +463,39 @@ class RooAbsReal : public RooAbsArg {
 
   //--------------------------------------------------------------------
 
- protected:
+  // Experimental AD feature functions
+  protected:
+  /// the variable that stores the result expression for the object.
+  std::string _adResult = "";
+
+  public:
+  virtual std::string translate(std::string &globalScope, std::vector<std::string> &preFuncDecls);
+  /// This function returns the variable/expression (as a std::string) that
+  /// the class stores its result in during code squashing. Allows for result
+  /// propagation up the compute graph.
+  virtual std::string getResult() { return _adResult; }
+  /// This function can be used to update the string that this class stores
+  /// its result into. Is beneficial for inputs or non-scalar variables that
+  /// might have varying index expressions.
+  virtual void updateResults(std::string newRes) { _adResult = newRes; }
+  /// A function to return if this class produces a loop. Loop producing
+  /// classes generally want to build the beginning of their loop before
+  /// visiting the subgraph.
+  virtual bool isLoopProducing() { return false; }
```

Comment:
```suggestion
```
This `isLoopProcessing` function is redundant with the existing `RooAbsArg::isReducerNode`, so you don't have to introduce `isLoopProcessing`.